client/app/services/file-management.js

```diff
@@ -0,0 +1,25 @@
+import Service, { inject as service } from '@ember/service';
+
+export default class FileManagementService extends Service {
+  @service
+  fileQueue;
+
+  // A hash of all File Managers in the app.
+  // There is one File Manager per editable Package.
+  // The key to access a File Manager is the respective
+  // package ID.
+  fileManagers = {}
+
+  createFileManager(
+    id = '',
+    existingFiles = [],
+  ) {
+    this.fileManagers[id] = {
+      existingFiles,
+      deleteFiles: [],
+      uploadFiles: this.fileQueue.create(id),
+    };
+
+    return this.fileManagers[id];
+  }
+}
```

Comment:
I really like that you wrapped this 3rd party file service inside a service that our app actually owns. This helps with testing a lot. I think we should make a habit of doing this :) great job